polyply/tests/test_big_smile_parsing.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+import networkx as nx
+from polyply.src.big_smile_parsing import (res_pattern_to_meta_mol,
+                                           tokenize_big_smile,
+                                           fragment_iter)
+
+@pytest.mark.parametrize('smile, nodes, edges',(
+                        # smiple linear seqeunce
+                        ("{[#PMA][#PEO][#PMA]}",
+                        ["PMA", "PEO", "PMA"],
+                        [(0, 1), (1, 2)]),
+                        # simple branched sequence
+                        ("{[#PMA][#PMA]([#PEO][#PEO])[#PMA]}",
+                        ["PMA", "PMA", "PEO", "PEO", "PMA"],
+                        [(0, 1), (1, 2), (2, 3), (1, 4)]),
+                        # simple sequence two branches
+                        ("{[#PMA][#PMA][#PMA]([#PEO][#PEO])([#CH3])[#PMA]}",
+                        ["PMA", "PMA", "PMA", "PEO", "PEO", "CH3", "PMA"],
+                        [(0, 1), (1, 2), (2, 3), (3, 4), (2, 5), (2, 6)]),
+                        # simple linear sequence with expansion
+                        ("{[#PMA]|3}",
+                        ["PMA", "PMA", "PMA"],
+                        [(0, 1), (1, 2)]),
+                        # simple branch expension
+                        ("{[#PMA]([#PEO][#PEO][#OHter])|2}",
+                        ["PMA", "PEO", "PEO", "OHter",
+                         "PMA", "PEO", "PEO", "OHter"],
+                        [(0, 1), (1, 2), (2, 3),
+                         (0, 4), (4, 5), (5, 6), (6, 7)]
+                         ),
+                        # nested branched with expansion
+                        ("{[#PMA]([#PEO]|3)|2}",
+                        ["PMA", "PEO", "PEO", "PEO",
+                         "PMA", "PEO", "PEO", "PEO"],
+                        [(0, 1), (1, 2), (2, 3),
+                         (0, 4), (4, 5), (5, 6), (6, 7)]
+                         )
+))
+def test_res_pattern_to_meta_mol(smile, nodes, edges):
```

Comment:
Add a case with nested branching: `xx(yy(z)y)x`

Reply:
done; there were some issues with nested branches and expansion. They are fixed now, but there are some pitfalls that need documentation in a documentation